packages/miniflare/src/workers/d1/dumpSql.ts

```diff
@@ -0,0 +1,152 @@
+// NOTE: this function duplicates the logic inside SQLite's shell.c.in as close
+// as possible, with any deviations noted.
+import { SqlStorage } from "@cloudflare/workers-types/experimental";
+
+export function* dumpSql(
```

Comment:
Is it worth building the array here instead of using a generator? It could simplify the code and would mean it wouldn't need to convert to an array in the caller. Or is there another reason why the sql return value needs to be in iterable?

Reply:
This is for parity with internal D1 code which uses a generator as it needs to stream the results out with backpressure. For Miniflare we keep the same code (duplicated as it didn't feel worth extracting a shared package for a single file that should very rarely change) but accumulate into an array in the caller.

Reply:
Though, in saying that, it would be nice to have a separate repo with a bunch of unit tests that cover regressions. At the moment that's inside D1's internal codebase, so I think I'll add a cleanup ticket to extract this out and unify the references. It would be nice to have this released in the mean time though.

Reply:
Makes sense, thank you!